File: src/main/java/fun/vs/fw/demo/domain/MerchantDirectory.java
package fun.vs.fw.demo.domain;


import fun.vs.fw.demo.domain.CategorizedTransaction.ExpenseCategory;
import fun.vs.fw.demo.domain.Transaction.MerchantCategoryCode;

@FunctionalInterface
public interface MerchantDirectory {
    ExpenseCategory getFor(MerchantCategoryCode code);
}

File: fvf4k/src/main/kotlin/fvf4k/demo/domain/MerchantDirectory.kt
package fvf4k.demo.domain

import arrow.core.raise.context.Raise
import fvf4k.demo.domain.model.ExpenseCategory
import fvf4k.demo.domain.model.MerchantCategoryCode


fun interface MerchantDirectory {
    context(_: Raise<ApplicationError>)
    fun getFor(mcc: MerchantCategoryCode): ExpenseCategory
}